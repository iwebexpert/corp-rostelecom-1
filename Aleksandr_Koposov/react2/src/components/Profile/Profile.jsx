import React, { useEffect, useState } from 'react'

import {
  Card,
  CardContent,
  CardActions,
  Badge,
  Icon,
  Avatar,
  TextField,
  Button,
  Dialog,
  DialogTitle,
  CircularProgress
} from '@material-ui/core'

import './Profile.scss'

export const Profile = (props) => {
  const { user, isLoading } = props
  const { id, name, age, about, avatar } = user || {}
  const [profileState, setProfile] = useState(
    {
      init: false,
      dialog: false,
      id,
      name,
      age,
      about,
      avatar
    }
  )

  useEffect(() => {
    setProfile({
      ...profileState,
      init: true,
      ...props.user
    })
  }, [])

  const onInputChange = (event) => {
    setProfile({
      ...profileState,
      [event.target.name]: event.target.value
    })
  }

  const saveChanges = () => {
    const { onSave } = props
    const { id, name, age, about, avatar } = profileState
    onSave({ id, name, age, about, avatar })
  }

  const toggleDialog = () => {
    setProfile({
      ...profileState,
      dialog: !profileState.dialog
    })
  }

  const avatarSize = {
    width: 128,
    height: 128
  }

  return (
    <Card className="profile" elevation={3}>
      <CardContent>
        <Badge
          overlap="circle"
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
          badgeContent={<Icon>edit</Icon>}
          onClick={toggleDialog}
        >
          {!isLoading ? (
            <Avatar
              style={avatarSize}
              alt="Аватар пользователя"
              src={profileState.avatar}
            />
          ) : (
              <Avatar style={avatarSize}>
                <CircularProgress />
              </Avatar>
            )}
        </Badge>
        <Dialog
          onClose={toggleDialog}
          open={profileState.dialog}
          className="avatar-dialog"
        >
          <DialogTitle>Выбор аватара</DialogTitle>
          <Card>
            <CardContent>
              <TextField
                name="avatar"
                label="Ссылка на аватар (можете изменять число в конце для теста)"
                variant="outlined"
                value={profileState.avatar}
                onChange={onInputChange}
                style={{ width: '550px' }}
              />
            </CardContent>
          </Card>
        </Dialog>
        {!isLoading ?
          (
            <div>
              <TextField
                name="name"
                label="Имя"
                variant="outlined"
                value={profileState.name}
                onChange={onInputChange}
              />
              <TextField
                name="age"
                label="Возраст"
                variant="outlined"
                type="number"
                value={profileState.age}
                onChange={onInputChange}
              />
              <TextField
                name="about"
                label="О себе"
                multiline
                variant="outlined"
                value={profileState.about}
                onChange={onInputChange}
              />
            </div>
          ) : (
            <div style={{
              width: '100%',
              textAlign: 'center'
            }}>
              <CircularProgress />
            </div>
          )}
      </CardContent>
      <CardActions>
        <Button
          color="primary"
          onClick={saveChanges}
        >Сохранить</Button>
      </CardActions>
    </Card>
  )
}
