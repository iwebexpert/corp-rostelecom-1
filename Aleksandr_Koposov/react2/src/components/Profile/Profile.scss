.profile {
  & .MuiBadge-badge {
    height: 36px;
    width: 36px;
    border-radius: 50%;
    background-color: grey;
    color: white;
    cursor: pointer;
  }

  & .MuiTextField-root {
    width: 100%;
    margin: 15px 0;
  }
}

.avatar-dialog {
  & .MuiTextField-root {
    max-width: 550px;
    width: 100%;
  }
}
