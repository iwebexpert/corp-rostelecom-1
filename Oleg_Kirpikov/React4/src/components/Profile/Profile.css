.profile {
    font-size: 2em;
    font-weight: bold;
    font-style: italic;
    text-align: center;
    background-color: rgb(186, 200, 228);
}
.panelButtons {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    
}