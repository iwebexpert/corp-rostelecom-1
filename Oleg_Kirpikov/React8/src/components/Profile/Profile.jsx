import React from 'react';
import { makeStyles } from '@material-ui/core';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';


const styles = makeStyles({
    root: {
        minWidth: 275,
    },
    bullet: {
        display: 'inline-block',
        margin: '0 2px',
        transform: 'scale(0.8)',
    },
    title: {
        fontSize: 14,
    },
    pos: {
        marginBottom: 12,
    },
});


export const Profile = (props) => {
    const classes = styles(props);

    const { profile } = props.profile;

    if (profile == undefined) {
        return (
            <Card variant="outlined">
            </Card>
        )
    };

    return (
        <Card variant="outlined" className={classes.root}>
            <CardHeader
                avatar={
                    <Avatar alt={` ${profile.firstName} ${profile.lastName} `} src={profile.avatar} />
                }
                title={` ${profile.firstName} ${profile.lastName} `}
            />
            <CardContent>
                <Typography className={classes.title} color="textSecondary" gutterBottom>
                    {profile.email}
                </Typography>
                <Typography variant="h5" component="h2">
                    {profile.description}
                </Typography>
                <Typography className={classes.pos} color="textSecondary">
                    {profile.title}
                </Typography>
                <Typography variant="body2" component="p">

                </Typography>
            </CardContent>
            <CardActions>
                <Button size="small">Закрыть</Button>
            </CardActions>
        </Card>
    )
}