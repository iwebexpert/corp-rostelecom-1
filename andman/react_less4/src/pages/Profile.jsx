import React from 'react';

export function ProfilePage() {
    return (
        <div>
            <h2>Profile page</h2>
            E-mail:  test@mail.ru <hr />
            Логин: Test1 <hr />
        </div>


    );
}