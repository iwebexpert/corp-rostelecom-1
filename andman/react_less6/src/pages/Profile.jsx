import React, { Component } from 'react';

export class ProfilePage extends Component {

    render() {
        const { profile } = this.props;
        //console.log(profile);
        return (
            <div>
                <h2>Profile page</h2>
        E - mail: {profile.email} <hr />
        Логин: {profile.name} < hr />
                <img src={profile.avatar} width="200px" height="200px" />
            </div >
        );
    };
}