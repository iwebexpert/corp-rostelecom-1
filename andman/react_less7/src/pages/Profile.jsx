import React, { Component } from 'react';

export class ProfilePage extends Component {

    render() {

        const { profile, isLoading, isError } = this.props;
        if (isError) {
            return (
                <div>Не удалось загрузить профиль...</div>
            );
        }
        if (isLoading) {
            return (
                <div>Profile loading...</div>
            );
        }
        return (
            <div>
                <h2>Profile page</h2>
        E - mail: {profile.email} <hr />
        Логин: {profile.name} < hr />
                <img src={profile.avatar} width="200px" height="200px" />
            </div >
        );

    };
}