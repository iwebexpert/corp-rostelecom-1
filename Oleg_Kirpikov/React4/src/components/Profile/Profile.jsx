import React, { Component } from 'react';
import { Grid } from '@material-ui/core';
import { TextField, Button } from '@material-ui/core';

import './Profile.css';

export class Profile extends Component {
    render() {
        return (
            <form>
                <Grid className="profile" >
                    <TextField label="Имя" name="author" variant="outlined" margin="normal" fullWidth />
                    <TextField label="Email" name="email" variant="outlined" margin="normal" fullWidth />
                    <TextField label="Пароль" name="passwd" variant="outlined" margin="normal" fullWidth />
                    <Grid item className="panelButtons">
                        <Button variant="contained" color="primary">
                            Сохранить
                        </Button>
                        <Button variant="contained" color="primary">
                            Закрыть
                        </Button>
                    </Grid>
                </Grid>
            </form>
        );
    }
}