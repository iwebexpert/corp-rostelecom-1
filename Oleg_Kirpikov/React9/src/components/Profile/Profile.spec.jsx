import React from 'react';
import { Profile } from './Profile';

describe('Profile component', () => {
    it('should render Profile component ', () => {
        const component = shallow(<Profile profile={{ author: 'WebDev', text: '1234' }} />);
        expect(component).toMatchSnapshot();
    });
});