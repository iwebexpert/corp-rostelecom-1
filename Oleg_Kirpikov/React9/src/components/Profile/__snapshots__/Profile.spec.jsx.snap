// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render Profile component  1`] = `
<WithStyles(ForwardRef(Card))
  variant="outlined"
/>
`;
